packages/contracts-bedrock/test/vendor/eas/SchemaRegistry.t.sol

```diff
@@ -0,0 +1,290 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import { Test } from "forge-std/Test.sol";
+import { console } from "forge-std/console.sol";
+import { SchemaRegistry, SchemaRecord } from "src/vendor/eas/SchemaRegistry.sol";
+import { ISchemaResolver } from "src/vendor/eas/resolver/ISchemaResolver.sol";
+
+contract SchemaRegistryTest is Test {
+    // State variables
+    SchemaRegistry registry;
+
+    // Events
+    event Registered(
+        bytes32 indexed uid,
+        address indexed registerer,
+        SchemaRecord schema
+    );
+
+    function setUp() public {
+        registry = new SchemaRegistry();
+    }
+
+    // Helper functions
+    function getUID(
+        string memory schema,
+        address resolver,
+        bool revocable
+    ) internal pure returns (bytes32) {
+        return keccak256(abi.encodePacked(schema, resolver, revocable));
+    }
+
+    // Should be equal to the current version of SchemaRegistry.sol in this repository
+    function testVersion() public view {
+        assertEq(registry.version(), "1.3.1-beta.1");
+    }
+
+    // Basic functionality tests
+    function testRegisterSchema() public {
```

Comment:
You can turn this into a fuzz test rather than hardcoding values, ie:

```solidity
    function testRegisterSchema(string memory _schema, address _resolver, bool _revocable) public {
```

then you can delete the hardcoded values. This improves the test coverage, foundry will run the test a ton of times (based on fuzz runs config in `foundry.toml`) so now it covers many values rather than just some values. If there is a value that doesn't work, you use `vm.assume(bool)` to filter out values, see the foundry docs and other uses of `vm.assume` in the codebase